pkg/actuators/machine/machine_scope.go

```diff
@@ -122,13 +122,15 @@ func (s *machineScope) getUserData() ([]byte, error) {
 		Name:      s.providerSpec.UserDataSecret.Name,
 	}
 
-	if err := s.client.Get(s.Context, objKey, userDataSecret); err != nil {
+	if err := s.client.Get(s.Context, objKey, userDataSecret); apierrors.IsNotFound(err) {
+		return nil, machineapierros.InvalidMachineConfiguration("userData secret is not found: %#v", objKey)
```

Comment:
We wouldn't typically use `%#v` to print something like this for an error message, it's not particularly user friendly, how about this instead?
```suggestion
		return nil, machineapierros.InvalidMachineConfiguration("userData secret is not found: %s/%s", objKey.Namespace, objKey.Name)
```